libcxx/include/string

```diff
@@ -2060,6 +2058,21 @@ private:
 #endif
   }
 
+  // Disable ASan annotations and enable them again when going out of scope. It is assumed that the string is in a valid
+  // state at that point, so `size()` can be called safely.
+  struct [[__nodiscard__]] __annotation_guard {
```

Comment:
Let's introduce an actual `__scope_guard` and then use it here. It's going to be more general and it will make this code easier to understand.

Reply:
This is #114867.